tests/test_main.py

```diff
@@ -1425,3 +1441,115 @@ class M(BaseModel):
         a: int
 
     get_type_hints(M.__config__)
+
+
+_5 = 5
+
+
+def _Annotated(*args):
+    return Annotated[args] if Annotated else None
```

Comment:
this is quite ugly, better to do something like `pytest.param(lambda: Annotated[int, 0], ...` if you do that you don't need the `if Annotated is None` case since the lambdas will only be evaluated when the function is called.

You should also split the cases with an error and the cases without so you don't need all the `nullcontext` logic.